data/campaigns/The_Hammer_of_Thursagan/scenarios/09_Forbidden_Forest.cfg

```diff
@@ -35,79 +35,14 @@
     # wmllint: validate-on
 
     # This tells the elves not to venture onto the plains
-#define AVOID_LEAVING_FOREST
-    [avoid]
-        x,y=3-4,6
-    [/avoid]
-    [avoid]
-        x,y=3-4,7
-    [/avoid]
-    [avoid]
-        x,y=1-3,8-9
-    [/avoid]
-    [avoid]
-        x,y=1-5,10-11
-    [/avoid]
-    [avoid]
-        x,y=1-2,12-14
-    [/avoid]
-    [avoid]
-        x,y=1-3,15
-    [/avoid]
-    [avoid]
-        x,y=1-4,16-18
-    [/avoid]
-    [avoid]
-        x,y=1-3,19-21
-    [/avoid]
-    [avoid]
-        x,y=1-2,22-23
-    [/avoid]
-    [avoid]
-        x,y=1,24
-    [/avoid]
-    [avoid]
-        x,y=1-2,25-27
-    [/avoid]
-    [avoid]
-        x,y=1-2,32-33
-    [/avoid]
-
-    [avoid]
-        x,y=35-40,1
-    [/avoid]
-    [avoid]
-        x,y=36-40,2
-    [/avoid]
-    [avoid]
-        x,y=39-40,3-4
-    [/avoid]
-    [avoid]
-        x,y=40,5-6
-    [/avoid]
-    [avoid]
-        x,y=39-40,7-10
-    [/avoid]
-    [avoid]
-        x,y=40,11-12
-    [/avoid]
-    [avoid]
-        x,y=39-40,13
-    [/avoid]
-    [avoid]
-        x,y=38-40,14
-    [/avoid]
-    [avoid]
-        x,y=39-40,15-19
-    [/avoid]
-#enddef
 
     # Set up the Staff of Righteous Flame.
     # Tell the AIs not to step on it so we don't have to deal
     # with the case where a forest unit picks it up.
-#define AVOID_TEMPLE
+#define AVOID
```

Comment:
I looked at a terrain-code selection and I'd have to [or] it with x,y because it would miss castles.

The 'best' method would actually be to load a terrain mask, query it, then discard it. But we can't do that.